### PR TITLE
EPMRPP-80662 || Browser parameter

### PR DESCRIPTION
**Pull request by tr1ble**

### CODE REVIEW

File: src/reporter.ts [hunk 1]
```diff
@@ -136,6 +139,7 @@ export class Reporter extends WDIOReporter {
       type: TYPES.STEP,
       codeRef,
       ...(this.options.cucumberNestedSteps && { hasStats: false }),
+      parameters: [this.sanitizedCapabilities && getBrowserParam(this.sanitizedCapabilities)],
```
Comment (AmsterGet) [inline src/reporter.ts:142]: I guess we should not report parameters in case of empty.

File: src/__tests__/onTestStart.spec.ts [hunk 2]
```diff
@@ -38,7 +38,12 @@ describe('onTestStart', () => {
 
     expect(reporter['client'].startTestItem).toBeCalledTimes(1);
     expect(reporter['client'].startTestItem).toBeCalledWith(
-      { name: testName, type: 'STEP', codeRef: '__test__/example.js/suite_name/test_name' },
+      {
+        name: testName,
+        type: 'STEP',
+        codeRef: '__test__/example.js/suite_name/test_name',
+        parameters: [undefined],
+      },
```
Comment (AmsterGet) [inline src/__tests__/onTestStart.spec.ts:46]: Please add test for existing parameters.

File: src/utils.ts [hunk 3]
```diff
@@ -183,3 +185,7 @@ export const limit = (val: any): any => {
     }
   }
 };
+
+export const getBrowserParam = (browser: string) => {
+  return { key: BROWSER_PARAM, value: browser };
+};
```
Comment (AmsterGet) [inline src/utils.ts:191]: Cover with tests or just provide this object directly in `onTestStart`.

**Review by AmsterGet — CHANGES_REQUESTED** [review]

File: src/reporter.ts [hunk 4]
```diff
@@ -131,11 +135,13 @@ export class Reporter extends WDIOReporter {
     const { title: name } = testStats;
     const ancestors = this.storage.getAllSuites();
     const codeRef = getCodeRef(this.testFilePath, name, ancestors);
+    const browser = this.sanitizedCapabilities;
     const testItemDataRQ = {
       name,
       type: TYPES.STEP,
       codeRef,
       ...(this.options.cucumberNestedSteps && { hasStats: false }),
+      parameters: browser && [{ key: BROWSER_PARAM, value: browser }],
```
Comment (chivekrodis) [inline src/reporter.ts:144]: I guess better write like this to omit key if `this.sanitizedCapabilities=false`
`...(this.sanitizedCapabilities && {parameters: [{ key: BROWSER_PARAM, value: browser }]})`
And variable `browser` looks unnecessary here

File: src/reporter.ts [hunk 5]
```diff
@@ -93,6 +96,7 @@ export class Reporter extends WDIOReporter {
     this.isMultiremote = runnerStats.isMultiremote;
     promiseErrorHandler(promise);
     this.tempLaunchId = tempId;
+    this.sanitizedCapabilities = runnerStats.sanitizedCapabilities;
```
Comment (chivekrodis) [inline src/reporter.ts:99]: Please move it higher close to `isMultiremote` to extract all params from `runnerStats` in the same place

File: src/reporter.ts [hunk 6]
```diff
@@ -38,6 +38,8 @@ import {
 import { CUCUMBER_TYPE, FILE_TYPES, LOG_LEVELS, RP_STATUSES, TYPES } from './constants';
 import { Attribute, FinishTestItem, LaunchObj, LogRQ, StartTestItem } from './models';
 
+const BROWSER_PARAM = 'browser';
```
Comment (chivekrodis) [inline src/reporter.ts:41]: Better move to constant file

**Review by chivekrodis — CHANGES_REQUESTED** [review]

**Review by chivekrodis — APPROVED** [review]

**Review by AmsterGet — APPROVED** [review]

**Pull request merged**